common/src/main/java/org/dash/wallet/common/WalletDataProvider.kt

```diff
@@ -22,19 +22,22 @@ import androidx.lifecycle.LiveData
 import org.bitcoinj.core.Address
 import org.bitcoinj.core.Coin
 import org.bitcoinj.core.Transaction
-import org.dash.wallet.common.data.ExchangeRate
+import org.dash.wallet.common.data.ExchangeRateData
 import org.dash.wallet.common.data.Resource
 
 interface WalletDataProvider {
 
-    fun freshReceiveAddress(): Address
+    fun currentReceiveAddress(): Address
```

Comment:
`currentReceiveAddress` is used by Uphold, I just left it as it is when merging two interfaces. But you're right, it might be better if Uphold used `freshReceiveAddress` as Liquid does. Maybe @HashEngineering knows the context.

Reply:
We should use `freshReceiveAddress` in Uphold as well.  We forgot to change it previously.

Otherwise it is possible that more than one transaction will be sent to the same address.  Ideally, we want all received transactions to be sent to separate addresses, at least as far as the app can control.